maven-settings-builder/src/main/java/org/apache/maven/settings/validation/DefaultSettingsValidator.java

```diff
@@ -60,7 +59,7 @@ public void validate(Settings settings, SettingsProblemCollector problems) {
             for (int i = 0; i < pluginGroups.size(); i++) {
                 String pluginGroup = pluginGroups.get(i).trim();
 
-                if (StringUtils.isBlank(pluginGroup)) {
+                if (pluginGroup.trim().isEmpty()) {
                     addViolation(
                             problems, Severity.ERROR, "pluginGroups.pluginGroup[" + i + "]", null, "must not be empty");
                 } else if (!ID_REGEX.matcher(pluginGroup).matches()) {
```

Comment:
This, the error message is misleading...

Reply:
I'm not sure I see how, but in any case that would be a pre-existing, unrelated issue. Please file a JIRA on this.

Reply:
The message says "empty", but test is for blank. We should check wether other spots validate and say then "empty". Those are two different things for me.

Reply:
Yes, and they aren't clear to me either. This is a problem with all the isEmpty, isBlank, isNullOrEmpty methods that litter utility libraries across Java, not just here but in Guava and many, many other libraries. They do not agree on what "empty", "blank", and similar terms mean. Sometimes "empty" strings can contain whitespace. Sometimes they can't. Sometimes they can be null. Sometimes they can't. It depends on which version of which library you're using. It's confusing and error prone. That's a big reason I want to get rid of all of them and stick exclusively to JDK methods.

In this case the string was already trimmed in line 60 so empty and blank are the same.

Reply:
I am included to drop the entire trim. The rest of the class does not trim at all. It should be consistent throughout elements.

Reply:
OK, trim dropped